lib/matplotlib/tests/test_legend.py

```diff
@@ -39,7 +40,7 @@ def test_legend_auto1():
     x = np.arange(100)
     ax.plot(x, 50 - x, 'o', label='y=1')
     ax.plot(x, x - 50, 'o', label='y=-1')
-    ax.legend(loc=0)
+    ax.legend()
```

Comment:
Tests don't like this...

Reply:
Interesting. Apparently, the legend is placed 'upper right' when no argument is given. Do tests have defaults different from the standard matplotlib defaults?

Reply:
I thought the default was 'best' instead of 'upper right'?

Reply:
maybe `style="mpl20"`?  Otherwise I think it uses classic...  But its for a test, so maybe being explicit is better than relying on the style choice?

Reply:
Ah, so the tests are still classic.

It's a bit unintuitive to have classic tests, because the code will behave different from more recent (v2) expectations. I see the point of not wanting to update all the tests just for the style though. Probably we should convert tests to mpl20 whenever we have to update the image anyway. That way, we'll get tests with mpl20 behavior in the long run without any additional overhead.

For now, I've put the explicit `loc='best'` back in.